orttraining/orttraining/training_api/module.h

```diff
@@ -21,6 +21,8 @@ struct Parameter {
 
   // Return the mutable data.
   OrtValue& Data() { return data_; }
+  Status CopyTo(const DataTransferManager* data_transfer_manager, OrtValue& data) const;
+  Status CopyFrom(const OrtValue& data, const DataTransferManager* data_transfer_manager);
```

Comment:
I recalled we had the API "CopyBufferToParameters" in Module, so am thinking should we add a similar API for the spcific parameter update in that class, beside the design consistency, we can also avoid passing the data transfer mgr explict in the copy API signature, because Module has the `train_sess_->GetDataTransferManager()`

Reply:
The `CheckpointState` has only two things (Properties and Parameters).

By moving access to all the parameters to the training session, we essentially are making the CheckpointState a redundant class and are preventing the isolation of responsibility between the training session and the checkpoint state.

I agree that we should have more consistency in the functions. But my feeling is that all the parameter related functions must be owned by the checkpoint state, and all the training related functions must be owned by the training session. Let me know your thoughts on this. I can make this change if you still feel that having the parameters accessible through the training session makes more sense.